src/nrniv/nrncore_write/io/nrncore_io.cpp

```diff
@@ -390,25 +397,84 @@ void write_nrnthread_task(const char* path, CellGroup* cgs)
     /// Writing the file with task, correspondent number of threads and list of correspondent first gids
     if (nrnmpi_myid == 0)
     {
+        // If append is false, begin a new files.dat (overwrite old if exists).
+        // If append is true, append groupids to existing files.dat.
+        //   Note: The number of groupids (2nd or 3rd line) has to be
+        //   overwritten wih the total number so far. To avoid copying
+        //   old to new, we allocate 10 chars for that number.
+
         std::stringstream ss;
         ss << path << "/files.dat";
 
         std::string filename = ss.str();
 
-        FILE *fp = fopen(filename.c_str(), "w");
-        if (!fp) {
-            hoc_execerror("nrnbbcore_write write_nrnthread_task could not open for writing:", filename.c_str());
+        FILE *fp = NULL;
+        if (append == false) { // start a new file
+            fp = fopen(filename.c_str(), "w");
+            if (!fp) {
+                hoc_execerror("nrncore_write write_nrnthread_task could not open for writing:", filename.c_str());
+            }
+        } else { // modify groupid number and append to existing file
+            fp = fopen(filename.c_str(), "r+");
+            if (!fp) {
+                hoc_execerror("nrncore_write write_nrnthread_task could not open for modifying:",  filename.c_str());
+            }
```

Comment:
this can be moved outside if/else with common message.

Reply:
Ok. The logic would need to be looked at, since hoc_execerror does no return.

Reply:
Returning to this, I don't know what common message is relevant. One is writing from scratch and the other is modifying.
Also, not only the name would be nice to have but also the ```strerror(errno)```

Reply:
I've changed the error messages to include the fact that this error happened in append mode.